src/main/java/dev/openfeature/javasdk/EvaluationContext.java

```diff
@@ -1,139 +1,31 @@
 package dev.openfeature.javasdk;
 
-import dev.openfeature.javasdk.internal.Pair;
-import lombok.*;
-
 import java.time.ZonedDateTime;
-import java.time.format.DateTimeFormatter;
-import java.util.HashMap;
-import java.util.Map;
+import java.util.List;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
 
-@ToString @EqualsAndHashCode
+@ToString @EqualsAndHashCode(callSuper = true)
 @SuppressWarnings("PMD.BeanMembersShouldSerialize")
-public class EvaluationContext {
+public class EvaluationContext extends Structure {
```

Comment:
I'm thinking about whether we should be extending structure here, rather than having an instance of Structure as an internal property. I think I have a preference for the latter?

One thing that makes me nervous about this EC/Structure merging is that Array type you added to the flag value types. We don't actually support returning arrays, and so parts of those are leaking between one another.

Thoughts?

Reply:
Splitting out the enum will address my concern for FVT. `@Delegate` looks like exactly what I was hoping for. Lombok is the best. Thank you!

Reply:
Lombok is incredible.

Reply:
I believe [this commit](https://github.com/open-feature/java-sdk/pull/51/commits/a8623253547411e94fe457adbb1fdd1a973779d7) resolves this concerns.

- encapsulates instead of inherits `Structure`, using lombok's `@Delegate`.
- adds new internal enum for `Structure` field types.